src/contracts/core/AllocationManager.sol

```diff
@@ -111,13 +111,14 @@ contract AllocationManager is
 
             // 5. Update state
             _updateAllocationInfo(params.operator, operatorSet.key(), strategy, info, allocation);
-            _updateMaxMagnitude(params.operator, strategy, info.maxMagnitude);
+            uint64 prevMaxMagnitude = _updateMaxMagnitude(params.operator, strategy, info.maxMagnitude);
 
-            // 6. Decrease operators shares in the DelegationManager
-            delegation.decreaseOperatorShares({
+            // 6. Decrease and burn operators shares in the DelegationManager
+            delegation.decreaseAndBurnOperatorShares({
                 operator: params.operator,
                 strategy: strategy,
-                wadSlashed: sharesWadSlashed
+                prevMaxMagnitude: prevMaxMagnitude,
+                newMaxMagnitude: info.maxMagnitude
             });
```

Comment:
I think we could just call this `burnOperatorShares` for brevity.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/881/commits/173c411ca2bc66853c4f417d09557a35dc5a4935